z390test/src/main/groovy/org/z390/test/z390Test.groovy

```diff
@@ -0,0 +1,266 @@
+            this.fileData[label] = outFile.text
+        }
+    }
+
+    static printFile(data, label) {
+        String linefeed = /\r\n|\n/
+        String[] lines = data.toString().split(linefeed)
+        println(("*" * 20) + " ${label} (${lines.length} lines) " + ("*" * 20))
+        lines.eachWithIndex{ String line, int lineNum ->
+            println("${String.format('%05d',lineNum)}  ${line}")
+        }
+    }
+
+    def printOutput(String... args) {
+        if (!printOutput) return
+        var label = args.join()
+        if (!label) {
+            label = "PRINT OUTPUT"
+        }
+        println("=-=-=-=-=-=-=-=-=-=-=  ${label} =-=-=-=-=-=-=-=-=-=-=")
+        printFile(this.stdout, 'stdout')
+        printFile(this.stderr, 'stderr')
+        this.fileData.each { fileExt, data ->
+            printFile(data, fileExt)
+        }
+    }
+
+    def mz390(Map kwargs=[:], String asmFilename, String... args) {
+        return asm(kwargs, asmFilename, args)
+    }
+
+    def asm(Map kwargs=[:], String asmFilename, String... args) {
+        /**
+         * Assemble
+         */
+        this.clean(asmFilename)
+        int rc = this.callZ390(asmFilename, 'mz390', args)
+        this.getOutput(asmFilename)
+        return rc
+    }
+
+    def lz390(Map kwargs=[:], String asmFilename, String... args) {
+        /**
+         * lz390
+         * !! NOTE !! Does not clean intermediate files
+         */
+        int rc = this.callZ390(asmFilename, 'lz390', args)
+        this.getOutput(asmFilename)
+        return rc
+    }
+
+    def asml(Map kwargs=[:], String asmFilename, String... args) {
+        /**
+         * Assemble and link
+         */
+        this.clean(asmFilename)
+        int rc = this.callZ390(asmFilename, 'mz390', args)
+        if (rc == 0) {
+            rc = this.callZ390(asmFilename, 'lz390', args)
+        }
+        this.getOutput(asmFilename)
+        return rc
+    }
+
+    def ez390(Map kwargs=[:], String asmFilename, String... args) {
+        /**
+         * ez390
+         * !! NOTE !! Does not clean intermediate files
+         */
+        int rc = this.callZ390(asmFilename, 'ez390', args)
+        this.getOutput(asmFilename)
+        return rc
+    }
+
+    def asmlg(Map kwargs=[:], String asmFilename, String... args) {
+        /**
+         * Assemble, link and go
+         */
+        this.clean(asmFilename)
+        int rc = this.callZ390(asmFilename, 'mz390', args)
+        if (rc == 0) {
+            rc = this.callZ390(asmFilename, 'lz390', args)
+            if (rc == 0) {
+                rc = this.callZ390(asmFilename, 'ez390', args)
+            }
+        }
+        this.getOutput(asmFilename)
+        return rc
+    }
+
+    def cblc(Map kwargs=[:], String cobFilename, String... args) {
+        /**
+         * cobol compile
+         */
+        this.clean(cobFilename)
+        int rc
+        rc = this.callZ390(cobFilename, 'zc390', *args)
+        if (rc != 0) return rc  // exit if issue
```

Comment:
The bash/cblc and bat\cblc files treat rc=4 as a warning and continue with the compile.
I suspect that exiting if rc is not 0 is okay here. Just pointing out the difference.

Reply:
It should be aligned to the bash/bat scripts so happy to update.
Will change to exit if not 0 or 4.